src/core/utils/color/ColorValue.ts

```diff
@@ -148,7 +148,8 @@ export class ColorValue {
    * Create a ColorValue from its 0xTTBBGGRR representation.
    */
   public static fromTbgr(tbgr: number): ColorValue {
-    return new ColorValue(tbgr);
+    const color = new ColorValue(tbgr);
+    return new ColorValue(tbgr, color.toHsvColor().h);
```

Comment:
What is this hack? You are creating a ColorValue and then creating another one from the same value?

Reply:
The first one creates the ColorValue but `_hue` is left undefined. The second creates a ColorValue with the correct hue.

Reply:
It was needed only for the tests to test if onChange was called with the expected color, but I removed this and updated tests to not need this anymore.

Reply:
The test looks weird now, with the `withoutHue` stuff. Any way around it? Maybe try initializing the component with rgb instead of hsl (there is no need for hue in a rgb test)
![image](https://user-images.githubusercontent.com/9084735/140337213-b0a134b3-086a-4c3a-ae01-355dd661dc6b.png)

Reply:
No, hue is being set in ColorInputPanel, after typing in the rgb input it creates the ColorValue, converts it to `HsvColor`, and then calls `applyHsvColorChange`. If I don't do the same in the test the hue values won't match.
![image](https://user-images.githubusercontent.com/88331924/140390851-4ae9fe78-a8f1-4bf4-8c6b-775dc3fcb85b.png)